// @looker/sdk-rtl/lib/esm/browserTransport.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BrowserTransport = exports.BrowserCryptoHash = void 0;
var _transport = require("./transport");
var _baseTransport = require("./baseTransport");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
class BrowserCryptoHash {
  arrayToHex(array) {
    return Array.from(array).map(b => b.toString(16).padStart(2, '0')).join('');
  }
  fromBase64(str) {
    return atob(str).split('').map(function (c) {
      return c.charCodeAt(0);
    });
  }
  secureRandom(byteCount) {
    var bytes = new Uint8Array(byteCount);
    window.crypto.getRandomValues(bytes);
    return this.arrayToHex(bytes);
  }
  sha256Hash(message) {
    return _asyncToGenerator(function* () {
      var msgUint8 = new TextEncoder().encode(message);
      var hashBuffer = yield window.crypto.subtle.digest('SHA-256', msgUint8);
      return (0, _transport.safeBase64)(new Uint8Array(hashBuffer));
    })();
  }
}
exports.BrowserCryptoHash = BrowserCryptoHash;
class BrowserTransport extends _baseTransport.BaseTransport {
  constructor(options) {
    super(options);
    this.options = options;
  }
  static supportsPerformance() {
    return !!(performance && performance.mark && performance.measure);
  }
  static get trackPerformance() {
    return this._trackPerf;
  }
  static set trackPerformance(value) {
    this._trackPerf = value && BrowserTransport.supportsPerformance();
  }
  static mark(name, tag) {
    if (this.trackPerformance) {
      var mark = "".concat(name, "-").concat(tag);
      performance.mark(mark);
      return mark;
    }
    return '';
  }
  static markName(url) {
    if (!this.trackPerformance) return '';
    var entries = performance.getEntriesByName(url, 'resource');
    if (entries.length > 0) {
      var last = entries[entries.length - 1];
      return "".concat(url, "-").concat(last.startTime);
    }
    return url;
  }
  static markStart(name) {
    return BrowserTransport.mark(name, BrowserTransport.startMark);
  }
  static markEnd(url, startName) {
    if (this.trackPerformance) {
      var measureName = this.markName(url);
      var end = BrowserTransport.mark(measureName, BrowserTransport.endMark);
      performance.measure(measureName, startName, end);
      performance.clearMarks(startName);
      performance.clearMarks(end);
      return measureName;
    }
    return '';
  }
  rawRequest(method, path, queryParams, body, authenticator, options) {
    var _this = this;
    return _asyncToGenerator(function* () {
      options = _objectSpread(_objectSpread({}, _this.options), options);
      var requestPath = _this.makeUrl(path, options, queryParams);
      var props = yield _this.initRequest(method, requestPath, body, authenticator, options);
      var req = fetch(props.url, props);
      var requestStarted = Date.now();
      var res = yield req;
      var responseCompleted = Date.now();
      var started = BrowserTransport.markStart(BrowserTransport.markName(requestPath));
      var contentType = String(res.headers.get('content-type'));
      var mode = (0, _transport.responseMode)(contentType);
      var responseBody = mode === _transport.ResponseMode.binary ? yield res.blob() : yield res.text();
      if (!('fromRequest' in options)) {
        BrowserTransport.markEnd(requestPath, started);
      }
      var headers = {};
      res.headers.forEach((value, key) => headers[key] = value);
      var response = {
        method,
        url: requestPath,
        body: responseBody,
        contentType,
        ok: true,
        statusCode: res.status,
        statusMessage: res.statusText,
        startMark: started,
        headers,
        requestStarted,
        responseCompleted
      };
      response.ok = _this.ok(response);
      return _this.observer ? _this.observer(response) : response;
    })();
  }
  parseResponse(res) {
    return _asyncToGenerator(function* () {
      var perfMark = res.startMark || '';
      if (!res.ok) {
        var _error = res.body;
        if (typeof _error === 'string') {
          try {
            _error = JSON.parse(_error);
          } catch (_unused) {
            _error = {
              message: "Request failed: ".concat(_error)
            };
          }
        }
        var response = {
          ok: false,
          error: _error
        };
        return response;
      }
      var value;
      var error;
      if (res.contentType.match(/application\/json/g)) {
        try {
          value = JSON.parse(yield res.body);
          BrowserTransport.markEnd(res.url, perfMark);
        } catch (err) {
          error = err;
          BrowserTransport.markEnd(res.url, perfMark);
        }
      } else if (res.contentType === 'text' || res.contentType.startsWith('text/')) {
        value = res.body.toString();
        BrowserTransport.markEnd(res.url, perfMark);
      } else {
        try {
          BrowserTransport.markEnd(res.url, perfMark);
          value = res.body;
        } catch (err) {
          BrowserTransport.markEnd(res.url, perfMark);
          error = err;
        }
      }
      var result;
      if (error) {
        result = {
          ok: false,
          error: error
        };
      } else {
        result = {
          ok: true,
          value
        };
      }
      return result;
    })();
  }
  request(method, path, queryParams, body, authenticator, options) {
    var _this2 = this;
    return _asyncToGenerator(function* () {
      try {
        if (BrowserTransport.trackPerformance) {
          options = _objectSpread(_objectSpread({}, options), {
            fromRequest: true
          });
        }
        var res = yield _this2.rawRequest(method, path, queryParams, body, authenticator, options);
        var result = yield _this2.parseResponse(res);
        return result;
      } catch (e) {
        if (!(0, _transport.isErrorLike)(e)) throw e;
        var error = {
          message: typeof e.message === 'string' ? e.message : "The SDK call was not successful. The error was '".concat(e, "'."),
          type: 'sdk_error'
        };
        return {
          error,
          ok: false
        };
      }
    })();
  }
  initRequest(method, path, body, authenticator, options) {
    var _this3 = this;
    return _asyncToGenerator(function* () {
      var _options;
      var agentTag = ((_options = options) === null || _options === void 0 ? void 0 : _options.agentTag) || _transport.agentPrefix;
      options = options ? _objectSpread(_objectSpread({}, _this3.options), options) : _this3.options;
      var headers = {
        [_transport.LookerAppId]: agentTag
      };
      if (options && options.headers) {
        Object.entries(options.headers).forEach(_ref => {
          var [key, val] = _ref;
          headers[key] = val;
        });
      }
      if (!body) {
        body = undefined;
      } else {
        if (typeof body !== 'string') {
          body = JSON.stringify(body);
          headers['Content-Type'] = 'application/json';
        }
      }
      var props = {
        body,
        credentials: 'same-origin',
        headers,
        method,
        url: path
      };
      if (authenticator) {
        props = yield authenticator(props);
      }
      return props;
    })();
  }
  stream(_callback, method, path, queryParams, body, authenticator, options) {
    var _this4 = this;
    return _asyncToGenerator(function* () {
      options = options ? _objectSpread(_objectSpread({}, _this4.options), options) : _this4.options;
      var requestPath = _this4.makeUrl(path, options, queryParams);
      var props = yield _this4.initRequest(method, requestPath, body, authenticator, options);
      (0, _transport.trace)("[stream] attempting to stream via download url", props);
      return Promise.reject(Error("Streaming for callback ".concat(props.method, " ").concat(props.requestPath, " is not implemented")));
    })();
  }
}
exports.BrowserTransport = BrowserTransport;
_defineProperty(BrowserTransport, "_trackPerf", false);
_defineProperty(BrowserTransport, "startMark", 'A');
_defineProperty(BrowserTransport, "endMark", 'B');
