// @looker/sdk-rtl/lib/esm/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.unquote = exports.matchModeString = exports.matchModeBinary = exports.matchCharsetUtf8 = exports.matchCharset = exports.isTrue = exports.isFalse = exports.boolDefault = void 0;
var matchCharset = ';.*charset=';
exports.matchCharset = matchCharset;
var matchCharsetUtf8 = "".concat(matchCharset, ".*\\butf-8\\b");
exports.matchCharsetUtf8 = matchCharsetUtf8;
var matchModeString = "(^application\\/.*(\\bjson\\b|\\bxml\\b|\\bsql\\b|\\bgraphql\\b|\\bjavascript\\b|\\bx-www-form-urlencoded\\b)|^text\\/|.*\\+xml\\b|".concat(matchCharset, ")");
exports.matchModeString = matchModeString;
var matchModeBinary = '^image\\/|^audio\\/|^video\\/|^font\\/|^application\\/|^multipart\\/';
exports.matchModeBinary = matchModeBinary;
var isTrue = value => /^(true|t|yes|y|1)$/i.test(value);
exports.isTrue = isTrue;
var isFalse = value => /^(false|f|no|n|0)$/i.test(value);
exports.isFalse = isFalse;
var boolDefault = function boolDefault(value) {
  var defaultBool = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
  if (isTrue(value)) return true;
  if (isFalse(value)) return false;
  return defaultBool;
};
exports.boolDefault = boolDefault;
var unquote = value => {
  if (!value) return '';
  if (/^['"`]/.test(value)) {
    var quote = value.substring(0, 1);
    if (value.endsWith(quote)) return value.substring(1, value.length - 1);
  }
  return value;
};
exports.unquote = unquote;
