// @looker/sdk-rtl/lib/apiMethods.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.functionalSdk = exports.APIMethods = void 0;
var _transport = require("./transport");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var functionalSdk = (authSession, apiVersion, sdkVersion) => {
  var sdk = new APIMethods(authSession, sdkVersion);
  sdk.apiVersion = apiVersion;
  sdk.apiPath = authSession.settings.base_url === '' ? '' : authSession.settings.base_url + '/api/' + apiVersion;
  return sdk;
};
exports.functionalSdk = functionalSdk;
class APIMethods {
  constructor(authSession, sdkVersion) {
    this.authSession = authSession;
    this.sdkVersion = sdkVersion;
    _defineProperty(this, "_apiPath", '');
    _defineProperty(this, "_apiVersion", '');
    if (!('agentTag' in authSession.settings && authSession.settings.agentTag)) {
      authSession.settings.agentTag = "".concat(_transport.agentPrefix, " ").concat(sdkVersion);
    }
  }
  get apiPath() {
    return this._apiPath;
  }
  set apiPath(value) {
    if (this._apiPath) {
      throw new Error("API Path is set to \"".concat(this._apiPath, "\" and cannot be reassigned"));
    }
    this._apiPath = value;
  }
  get apiVersion() {
    return this._apiVersion;
  }
  set apiVersion(value) {
    if (this._apiVersion) {
      throw new Error("API Version is set to \"".concat(this._apiVersion, "\" and cannot be reassigned"));
    }
    this._apiVersion = value;
  }
  static create(type, authSession) {
    return new type(authSession);
  }
  ok(promise) {
    return _asyncToGenerator(function* () {
      return (0, _transport.sdkOk)(promise);
    })();
  }
  makePath(path, options, authenticator) {
    if (path.match(/^(http:\/\/|https:\/\/)/gi)) return path;
    var base = authenticator ? this.apiPath : options.base_url;
    return "".concat(base).concat(path);
  }
  authRequest(method, path, queryParams, body, options) {
    var _this = this;
    return _asyncToGenerator(function* () {
      options = _objectSpread(_objectSpread({}, _this.authSession.settings), options);
      var authenticator = init => {
        return _this.authSession.authenticate(init);
      };
      path = _this.makePath(path, options, authenticator);
      return _this.authSession.transport.request(method, path, queryParams, body, authenticator, options);
    })();
  }
  authStream(callback, method, path, queryParams, body, options) {
    var _this2 = this;
    return _asyncToGenerator(function* () {
      options = _objectSpread(_objectSpread({}, _this2.authSession.settings), options);
      var authenticator = init => {
        return _this2.authSession.authenticate(init);
      };
      path = _this2.makePath(path, options, authenticator);
      return _this2.authSession.transport.stream(callback, method, path, queryParams, body, init => {
        return _this2.authSession.authenticate(init);
      }, options);
    })();
  }
  get(path, queryParams, body, options) {
    var _this3 = this;
    return _asyncToGenerator(function* () {
      return _this3.authRequest('GET', path, queryParams, body, options);
    })();
  }
  head(path, queryParams, body, options) {
    var _this4 = this;
    return _asyncToGenerator(function* () {
      return _this4.authRequest('HEAD', path, queryParams, body, options);
    })();
  }
  delete(path, queryParams, body, options) {
    var _this5 = this;
    return _asyncToGenerator(function* () {
      return _this5.authRequest('DELETE', path, queryParams, body, options);
    })();
  }
  post(path, queryParams, body, options) {
    var _this6 = this;
    return _asyncToGenerator(function* () {
      return _this6.authRequest('POST', path, queryParams, body, options);
    })();
  }
  put(path, queryParams, body, options) {
    var _this7 = this;
    return _asyncToGenerator(function* () {
      return _this7.authRequest('PUT', path, queryParams, body, options);
    })();
  }
  patch(path, queryParams, body, options) {
    var _this8 = this;
    return _asyncToGenerator(function* () {
      return _this8.authRequest('PATCH', path, queryParams, body, options);
    })();
  }
}
exports.APIMethods = APIMethods;
