// @looker/sdk/lib/esm/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  models31: true,
  functionalSdk31: true,
  funcs31: true,
  ILooker31SDK: true,
  Looker31SDK: true,
  Looker31SDKStream: true,
  ILooker40SDK: true,
  Looker40SDK: true,
  Looker40SDKStream: true,
  functionalSdk40: true,
  sdkVersion: true,
  environmentPrefix: true,
  LookerExtensionSDK: true,
  BrowserSettings: true,
  LookerBrowserSDK: true
};
Object.defineProperty(exports, "BrowserSettings", {
  enumerable: true,
  get: function get() {
    return _browserSdk.BrowserSettings;
  }
});
Object.defineProperty(exports, "ILooker31SDK", {
  enumerable: true,
  get: function get() {
    return _methodsInterface.ILooker31SDK;
  }
});
Object.defineProperty(exports, "ILooker40SDK", {
  enumerable: true,
  get: function get() {
    return _methodsInterface2.ILooker40SDK;
  }
});
Object.defineProperty(exports, "Looker31SDK", {
  enumerable: true,
  get: function get() {
    return _methods.Looker31SDK;
  }
});
Object.defineProperty(exports, "Looker31SDKStream", {
  enumerable: true,
  get: function get() {
    return _streams.Looker31SDKStream;
  }
});
Object.defineProperty(exports, "Looker40SDK", {
  enumerable: true,
  get: function get() {
    return _methods2.Looker40SDK;
  }
});
Object.defineProperty(exports, "Looker40SDKStream", {
  enumerable: true,
  get: function get() {
    return _streams2.Looker40SDKStream;
  }
});
Object.defineProperty(exports, "LookerBrowserSDK", {
  enumerable: true,
  get: function get() {
    return _browserSdk.LookerBrowserSDK;
  }
});
Object.defineProperty(exports, "LookerExtensionSDK", {
  enumerable: true,
  get: function get() {
    return _extensionSdk.LookerExtensionSDK;
  }
});
Object.defineProperty(exports, "environmentPrefix", {
  enumerable: true,
  get: function get() {
    return _constants.environmentPrefix;
  }
});
exports.funcs31 = void 0;
Object.defineProperty(exports, "functionalSdk31", {
  enumerable: true,
  get: function get() {
    return funcs31.functionalSdk31;
  }
});
Object.defineProperty(exports, "functionalSdk40", {
  enumerable: true,
  get: function get() {
    return _funcs2.functionalSdk40;
  }
});
exports.models31 = void 0;
Object.defineProperty(exports, "sdkVersion", {
  enumerable: true,
  get: function get() {
    return _constants.sdkVersion;
  }
});
var models31 = _interopRequireWildcard(require("./3.1/models"));
exports.models31 = models31;
var funcs31 = _interopRequireWildcard(require("./3.1/funcs"));
exports.funcs31 = funcs31;
var _methodsInterface = require("./3.1/methodsInterface");
var _methods = require("./3.1/methods");
var _streams = require("./3.1/streams");
var _methodsInterface2 = require("./4.0/methodsInterface");
var _methods2 = require("./4.0/methods");
var _streams2 = require("./4.0/streams");
var _funcs2 = require("./4.0/funcs");
Object.keys(_funcs2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _funcs2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _funcs2[key];
    }
  });
});
var _models2 = require("./4.0/models");
Object.keys(_models2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _models2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _models2[key];
    }
  });
});
var _constants = require("./constants");
var _extensionSdk = require("./extensionSdk");
var _browserSdk = require("./browserSdk");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
